File: workflow/plugins/go/var/var_get/var_get.go
// Package var_get provides the variable get plugin.
package var_get

import (
	plugin "metabuilder/workflow/plugins/go"
)

// Run retrieves a variable from the workflow store.
func Run(runtime *plugin.Runtime, inputs map[string]interface{}) (map[string]interface{}, error) {
	key, ok := inputs["key"].(string)
	if !ok {
		return map[string]interface{}{
			"result": nil,
			"exists": false,
			"error":  "key is required",
		}, nil
	}

	defaultVal := inputs["default"]

	value, exists := runtime.Store[key]
	if !exists {
		value = defaultVal
	}

	return map[string]interface{}{
		"result": value,
		"exists": exists,
	}, nil
}

File: workflow/plugins/python/var/var_get/var_get.py
"""Workflow plugin: get variable from workflow store."""


def run(runtime, inputs):
    """Get variable value from workflow store."""
    key = inputs.get("key")
    default = inputs.get("default")

    if key is None:
        return {"result": default, "exists": False, "error": "key is required"}

    exists = key in runtime.store
    value = runtime.store.get(key, default)

    return {"result": value, "exists": exists}
